src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/adsbase/helpers/AdsComparison.java

```diff
@@ -111,17 +113,17 @@ public boolean isAdVisible(final WebElement element, final String selector,
       wait.until(new ExpectedCondition<Object>() {
         @Override
         public Object apply(WebDriver driver) {
-          File adImg = shooter.captureWebElement(element, driver);
-          PageObjectLogging.log("ScreenshotsComparison", "Ad image in " + selector, true, driver);
-          boolean areFilesTheSame = imageComparison.areFilesTheSame(backgroundImg, adImg);
-          adImg.delete();
-          return !areFilesTheSame;
+          BufferedImage adImg = shooter.takeScreenshot(element, driver);
```

Comment:
@ludwikkazmierczak, which calculation an `takeScreenshot` ?
AdsComparison class is exactly for checking such things like whether an ad visible or not.
If you are talking about getting variance, but it's already in the helper images class.